samples/wifi/sta/src/main.c

```diff
@@ -0,0 +1,257 @@
+
+		return -ENOEXEC;
+	}
+
+	printk("Status: successful\n");
+	printk("==================\n");
+	printk("State: %s\n", wifi_state_txt(status.state));
+
+	if (status.state >= WIFI_STATE_ASSOCIATED) {
+		uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+
+		printk("Interface Mode: %s\n",
+				wifi_mode_txt(status.iface_mode));
+		printk("Link Mode: %s\n",
+				wifi_link_mode_txt(status.link_mode));
+		printk("SSID: %-32s\n", status.ssid);
+		printk("BSSID: %s\n",
+				status.bssid ? net_sprint_ll_addr_buf(status.bssid,
+					WIFI_MAC_ADDR_LEN, mac_string_buf,
+					sizeof(mac_string_buf)) :
+				"");
+		printk("Band: %s\n",
+				wifi_band_txt(status.band));
+		printk("Channel: %d\n", status.channel);
+		printk("Security: %s\n",
+				wifi_security_txt(status.security));
+		printk("MFP: %s\n",
+				wifi_mfp_txt(status.mfp));
+		printk("RSSI: %d\n", status.rssi);
+	}
+
+
+	return 0;
+}
+
+static void handle_wifi_connect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *) cb->info;
+
+	if (status->status) {
+		printk("Connection request failed (%d)\n", status->status);
+	} else {
+		printk("Connected\n");
+	}
+
+	cmd_wifi_status();
+
+	context.connecting = false;
+}
+
+static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *) cb->info;
+
+	if (context.disconnecting) {
+		printk("Disconnection request %s (%d)\n",
+				status->status ? "failed" : "done",
+				status->status);
+		context.disconnecting = false;
+	} else {
+		printk("Disconnected\n");
+	}
+
+	cmd_wifi_status();
+}
+
+static void wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+		uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+		case NET_EVENT_WIFI_CONNECT_RESULT:
+			handle_wifi_connect_result(cb);
+			break;
+		case NET_EVENT_WIFI_DISCONNECT_RESULT:
+			handle_wifi_disconnect_result(cb);
+			break;
+		default:
+			break;
+	}
+}
+
+static void print_dhcp_ip(struct net_mgmt_event_callback *cb)
+{
+	/* Get DHCP info from struct net_if_dhcpv4 and print */
+	const struct net_if_dhcpv4 *dhcpv4 = cb->info;
+	const struct in_addr *addr = &dhcpv4->requested_ip;
+	char dhcp_info[128];
+
+	net_addr_ntop(AF_INET, addr, &dhcp_info, sizeof(dhcp_info));
+
+	printk("IP address: %s\n", dhcp_info);
+}
+
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+		uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+		case NET_EVENT_IPV4_DHCP_BOUND:
+			print_dhcp_ip(cb);
+			break;
+		default:
+			break;
+	}
+}
+
+static int __wifi_args_to_params(struct wifi_connect_req_params *params)
+{
+	params->timeout = SYS_FOREVER_MS;
+
+	/* SSID */
+	params->ssid = CONFIG_STA_SAMPLE_SSID;
+	params->ssid_length = strlen(params->ssid);
+
+	params->security = CONFIG_STA_KEY_MGMT_SELECT;
+	if (params->security) {
+		params->psk = CONFIG_STA_SAMPLE_PWD;
+		params->psk_length = strlen(params->psk);
```

Comment:
This will work, but for WPA3 we ideally need to use `sae_password`

Reply:
> This will work, but for WPA3 we ideally need to use `sae_password`

I guess, we just need to update "struct wifi_connect_req_params " fields for connect.